code/ModelAdmin.php

```diff
@@ -188,6 +189,17 @@ public function getEditForm($id = null, $fields = null)
                 ->addComponents(new GridFieldPrintButton('buttons-before-left'))
         );
 
+        $listField->getConfig()->removeComponentsByType(GridFieldFilterHeader::class);
```

Comment:
I think it would be prudent to comment why we're doing this and mention it's for BC until 5.0.

Reply:
Yep, done.